au3/libraries/lib-wave-track/TimeStretching.cpp

```diff
@@ -15,11 +15,41 @@
 #include "TempoChange.h"
 #include "UserException.h"
 #include "WaveClip.h"
+#include "realfn.h"
 #include <algorithm>
 
 const TranslatableString TimeStretching::defaultStretchRenderingTitle =
    XO("Pre-processing");
 
+namespace {
+void trimRightOverlapping(WaveTrack &track) {
```

Comment:
there is a function `GetClipsIntersecting` that returns a list of clips in a given region that might help here

Reply:
Yes, but `GetClipsIntersecting` checks if there are clips within given t0<->t1 boundaries and in this case we need to check if any pair of clips overlap each other and calculate by how much. I just simply follow the same rules as the other overlapping functions